google-cloud-bigquery/src/main/java/com/google/cloud/bigquery/StandardTableDefinition.java

```diff
@@ -272,7 +272,27 @@ static StandardTableDefinition fromPb(Table tablePb) {
       builder.setStreamingBuffer(StreamingBuffer.fromPb(tablePb.getStreamingBuffer()));
     }
     if (tablePb.getTimePartitioning() != null) {
-      builder.setTimePartitioning(TimePartitioning.fromPb(tablePb.getTimePartitioning()));
+      try {
+        builder.setTimePartitioning(TimePartitioning.fromPb(tablePb.getTimePartitioning()));
+      } catch (IllegalArgumentException e) {
+        throw new IllegalArgumentException(
+            "Got unexpected time partitioning "
+                + tablePb.getTimePartitioning()
+                + " in table "
+                + tablePb.getTableReference(),
+            e);
+      } catch (NullPointerException e) {
+        try {
+          throw new NullPointerException(
+                  "Got unexpected time partitioning "
+                      + tablePb.getTimePartitioning()
+                      + " in table "
+                      + tablePb.getTableReference())
+              .initCause(e);
+        } catch (Throwable throwable) {
+          throwable.printStackTrace();
```

Comment:
Are we expecting to get errors when constructing the exception?

Reply:
Or is this due to the `initCause`? If so, let's add a comment.

Reply:
due to initCause - added comment

Reply:
No - this is for .initiCause